strawberry/experimental/pydantic/object_type.py

```diff
@@ -83,6 +83,7 @@ def type(
     description: Optional[str] = None,
     directives: Optional[Sequence[StrawberrySchemaDirective]] = (),
     all_fields: bool = False,
+    gql_uses_alias: bool = True,
```

Comment:
couldn't think of a better name that isn't too long. any ideas?

Reply:
maybe we could use just `use_alias`?

let me /cc @Matt343 as he's probably using the pydantic integration, I don't use it at the moment so I'm not 100% sure if we should do this :)

Reply:
I'm just wondering if this option should be specified at the field level instead of the object level, since you might want to use an alias in one place but not another. I'm also not clear on what affect an alias has if you turn this feature off, where would it actually be used?
For the first issue related to `strawberry.auto`, I don't think that's the intended behavior so maybe we should be open to a bugfix there?

Reply:
>  I'm also not clear on what affect an alias has if you turn this feature off, where would it actually be used?

Not sure if i understand right. But [the alias is the pydantic alias i'm referring to](https://pydantic-docs.helpmanual.io/usage/model_config/#alias-precedence)


> I'm just wondering if this option should be specified at the field level instead of the object level, since you might want to use an alias in one place but not another. 

interesting. how could we support this?
we could have some Annotation on the pydantic model. But idk how well that would work with pydantic. And personally i don't like mixing strawberry logic in my pydantic models layer.
maybe we can do something like this?

```python
@strawberry.experimental.pydantic.type(
    UserModel
)
class User:
    id: strawberry.auto(rename_to="someOtherName")
    another_field: strawberry.auto
```


> For the first issue related to strawberry.auto, I don't think that's the intended behavior so maybe we should be open to a bugfix there?

cool! though it the default behaviour for sometime so idk if theres people relying on it. and schema changes aren't so obvious :(

Reply:
Ah I see, I thought we were talking about strawberry aliases here. This makes a lot of sense now, and I can see how the current behavior is pretty confusing. What you're actually running into is an undocumented feature/bug in the pydantic integration that exists when you define a field that has the same name as a pydantic field but doesn't use the `auto` type. What that does is sort of copy the field from pydantic but not all the way, which leads to very strange behaviors. I'm not sure what we should do here, but it's certainly not what we do today, even if that behavior is sometimes useful by accident.

As for where to put the setting, I like offering the field level if possible but maybe what you have now is a good starting point. You wouldn't be able to do `strawberry.auto(rename_to="someOtherName")` afaik due to the way that type annotations need to work, but we could potentially add something like:

```
@strawberry.experimental.pydantic.type(
    UserModel
)
class User:
    id: strawberry.auto = strawberry.field(alias="someOtherName")
    another_field: strawberry.auto
```

This would be some work though, so I feel like it might be good as a follow on feature rather than a blocker now, I don't see any reason not to have the object level config anyway.

Reply:
great! seems like we're on the same page. sorry didn't explain that it was a pydantic alias earlier :) 
@patrick91 could we add the object level config first?

Reply:
Hi all, matt suggested that we can have an field level alias by `strawberry.field(alias="someOtherName")`.
i think thats reasonable and something we can support in a future MR.

for this MR, would like to continue forward to support a configuration at the decorator level.

 i have changed the decorator parameter to `use_pydantic_alias` to differentiate the pydantic alias from a possible future strawberry.field(alias="someOtherName") parameter.